version/version.go

```diff
@@ -0,0 +1,6 @@
+// Package version handles k6 versioning
+package version
```

Comment:
I'm curious if we really need the package for that? :thinking: 

How about moving this to the root (main package) and just access it as `go.k6.io/k6.Version` ?

Reply:
Unfortunately, If I understand right your question, AFAIK, it is not possible to import a `main` package from another library.

Reply:
I kind of think that there should be a way without referring to the `main` package, but using `go.k6.io/k6` on the other side quickly checking there immediately cycle imports appears so maybe not this time

Reply:
[Packages and commands in the same repository](https://go.dev/doc/modules/layout#packages-and-commands-in-the-same-repository) section of this `godoc` page explains what we should do for supporting a root package as `k6`.

TL;DR we should move the `main.go` file under some other package. Here a draft pull request experimenting with the idea. https://github.com/grafana/k6/pull/4560.

The issue is that it would break anyone building from master. Our CI included.

Reply:
Passing through, just to say that whatever decision we make in regards to this, it should be now or "never" (till v2).
Perhaps it worth a larger discussion with the entire group.

Reply:
> Perhaps it worth a larger discussion with the entire group.

@joanlopez do we really have an option here? I think the cost-benefit here is really unbalanced on the costs.

Because it would introduce a lot of breaking changes on critical commands like `go build` / `go install`, but also the UX introduced for the `go install` command is going to be less effective. `go install go.k6.io/k6@latest` is easier than `go install go.k6.io/k6/cmd/k6@latest`.

In terms of benefit, we don't have a real case except the one we are discussing, which has already a fine work-around with the `version` package.

Reply:
Ah, no, no. I just commented out what I think should be next steps in case we want to explore that path, which is what I got from your draft PR.

But not something I really want to push for.


Indeed, my first reaction with the `version` package was similar to @olegbespalov's one, but I didn't ever commented it because I suspected a bunch of difficulties.